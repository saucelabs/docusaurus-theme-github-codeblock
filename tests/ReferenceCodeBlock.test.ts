import { parseCustomization, parseReference, codeReducer } from '../src/theme/ReferenceCodeBlock/index'

test('should parse GitHub reference properly', () => {
    expect(parseReference('https://github.com/saucelabs/docusaurus-theme-github-codeblock/blob/main/src/theme/ReferenceCodeBlock/index.tsx'))
        .toMatchSnapshot()
    expect(parseReference('https://github.com/saucelabs/docusaurus-theme-github-codeblock/blob/main/src/theme/ReferenceCodeBlock/index.tsx#L105-L108'))
        .toMatchSnapshot()
})

test('should use custom reference link text', () => {
    expect(parseCustomization('https://github.com/saucelabs/docusaurus-theme-github-codeblock/blob/main/src/theme/ReferenceCodeBlock/index.tsx?referenceLinkText="Sample text"'))
        .toMatchSnapshot()
    expect(parseCustomization('https://github.com/saucelabs/docusaurus-theme-github-codeblock/blob/main/src/theme/ReferenceCodeBlock/index.tsx?referenceLinkText="Sample text"&title="Sample title"'))
        .toMatchSnapshot()
    expect(parseCustomization('https://github.com/saucelabs/docusaurus-theme-github-codeblock/blob/main/src/theme/ReferenceCodeBlock/index.tsx?referenceLinkText="Sample text"&title="Sample title"&customStyling'))
        .toMatchSnapshot()
})

test('codeReducer', () => {
    const prevState = { foo: 'bar' }
    expect(codeReducer(prevState, { type: 'reset', value: '' })).toMatchSnapshot()
    expect(codeReducer(prevState, { type: 'loading', value: '' })).toMatchSnapshot()
    expect(codeReducer(prevState, { type: 'loaded', value: 'foobar' })).toMatchSnapshot()
    expect(codeReducer(prevState, { type: 'error', value: 'ups' })).toMatchSnapshot()
    // @ts-expect-error
    expect(codeReducer(prevState, { type: 'unknown', value: '' })).toEqual(prevState)
})

