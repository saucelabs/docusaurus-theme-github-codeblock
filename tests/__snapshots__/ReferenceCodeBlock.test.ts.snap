// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`codeReducer 1`] = `
Object {
  "code": "loading...",
  "error": null,
  "loading": null,
}
`;

exports[`codeReducer 2`] = `
Object {
  "foo": "bar",
  "loading": true,
}
`;

exports[`codeReducer 3`] = `
Object {
  "code": "foobar",
  "foo": "bar",
  "loading": false,
}
`;

exports[`codeReducer 4`] = `
Object {
  "error": "ups",
  "foo": "bar",
  "loading": false,
}
`;

exports[`should parse GitHub reference properly 1`] = `
Object {
  "fromLine": 0,
  "title": "src/theme/ReferenceCodeBlock/index.ts",
  "toLine": Infinity,
  "url": "https://raw.githubusercontent.com/saucelabs/docusaurus-theme-github-codeblock/main/src/theme/ReferenceCodeBlock/index.ts",
}
`;

exports[`should parse GitHub reference properly 2`] = `
Object {
  "fromLine": 104,
  "title": "src/theme/ReferenceCodeBlock/index.tsx",
  "toLine": 9,
  "url": "https://raw.githubusercontent.com/saucelabs/docusaurus-theme-github-codeblock/main/src/theme/ReferenceCodeBlock/index.tsx",
}
`;

exports[`should use custom reference link text 1`] = `
Object {
  "linkText": "\\"Sample text\\"",
  "noteStyling": Object {
    "color": "#0E75DD",
    "fontSize": ".9em",
    "fontWeight": 600,
    "paddingBottom": "13px",
    "textAlign": "center",
    "textDecoration": "underline",
  },
  "title": null,
  "useCustomStyling": false,
}
`;

exports[`should use custom reference link text 2`] = `
Object {
  "linkText": "\\"Sample text\\"",
  "noteStyling": Object {
    "color": "#0E75DD",
    "fontSize": ".9em",
    "fontWeight": 600,
    "paddingBottom": "13px",
    "textAlign": "center",
    "textDecoration": "underline",
  },
  "title": "\\"Sample title\\"",
  "useCustomStyling": false,
}
`;

exports[`should use custom reference link text 3`] = `
Object {
  "linkText": "\\"Sample text\\"",
  "noteStyling": Object {},
  "title": "\\"Sample title\\"",
  "useCustomStyling": true,
}
`;
